xpk.py

```diff
@@ -243,358 +242,408 @@
         command: [ "sleep", "inf" ]
 """
 
+# @dataclass
+# class AcceleratorType(Enum):
+#     TPU = 1
+#     GPU = 2
+#     # CPU = 3,
+
+AcceleratorType = {
+  'TPU': 1,
+  'GPU': 2
+}
+
+@dataclass
+class AcceleratorCharacteristics:
+  resource_type: str
+  accelerator_label: int
+  machine_label: str
+
+AcceleratorTypeToAcceleratorCharacteristics = {
+     #tpu
+      AcceleratorType['TPU']: AcceleratorCharacteristics(
+      'google.com/tpu', 'cloud.google.com/gke-tpu-accelerator', ' cloud.google.com/gke-tpu-topology'
+      ),
+     #gpu
+      AcceleratorType['GPU']: AcceleratorCharacteristics(
+      'nvidia.com/gpu', 'cloud.google.com/gke-accelerator', ' cloud.google.com/gce-machine-type'
+      )
+}
+
+
 @dataclass
 class SystemCharacteristics:
   topology: str
   vms_per_slice: int
   gke_accelerator: str
   gce_machine_type: str
   chips_per_vm: int
+  accelerator_type: AcceleratorType
 
 ################### Subcommand Helper Functions #############
 """ !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 IF YOU MODIFY THE BELOW UserFacingNameToSystemCharacteristics MAP YOU SHOULD ALSO ADD CORRESPONDING
 MODIFICATIONS TO UserFacingNameToSystemCharacteristics IN MaxText/accelerator_to_spec_map.py !!!!! """
 # vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
-UserFacingNameToSystemCharacteristics = {
+GpuUserFacingNameToSystemCharacteristics = {
```

Comment:
We can condense our system characteristics to one dictionary. Since we store the accelerator type as one of the values, that tells us which type of device it is.

Reply:
This will also reduce the if statements in our code!